### DISCUSSION TITLE
NuGet Package Issue in OpenGlab UnitTest Project

### DISCUSSION THREAD

**Discussion by Alisl001** [MuhammadSulaiman001/opengl-lab#21] [Q&A]
Hi @MuhammadSulaiman001 

Hope you're well. 

When attempting to build the solution for the OpenGlab in the UnitTest project, I encountered the following error:
 `"OpenGLab\UnitTests\UnitTests.vcxproj" (default target) (7) ->
(EnsureNuGetPackageBuildImports target) ->  error : This project references NuGet package(s) that are missing on this computer. Use NuGet Package Restore to download them. For more information, see http://go.microsoft.com/fwlink/?LinkID=322105. The missing file is ..\..\..\..\..\..\.nuget\packages\Microsoft.googletest.v140.windesktop.msvcstl.static.rt-dyn.1.8.1.7\build\native\Microsoft.googletest.v140.windesktop.msvcstl.static.rt-dyn.targets."
` 
I visited the provided link in the error message and explored various avenues to resolve this issue, but unfortunately, none of the methods proved successful.

 In Visual Studio, I tried to install the required package using both the Package Manager Console and PowerShell with the command:
 `NuGet\Install-Package Microsoft.googletest.v140.windesktop.msvcstl.static.rt-dyn -Version 1.8.1.7
`the console returned an error indicating that no project was found to install the package. I've exhausted my efforts in attempting to resolve this via the package installation methods.

It's worth noting that the LearnOpenGl repository is free of errors, and I can successfully work with it. The issue appears to be specific to the OpenGlab repository.

I would greatly appreciate your insights and assistance in resolving this error. Thank you for your time and support.

**Comment by MuhammadSulaiman001** [comment]
Hi @Alisl001

Thanks for the report.

> In Visual Studio, I tried to install the required package using both the Package Manager Console and PowerShell with the command:
NuGet\Install-Package Microsoft.googletest.v140.windesktop.msvcstl.static.rt-dyn -Version 1.8.1.7 the console returned an error indicating that no project was found to install the package. I've exhausted my efforts in attempting to resolve this via the package installation methods.

Although this package should be installed automatically when building the project (unless you have internet or VPN issues).. If you want to install the package manually, `Install-Package ..` should be run while the working directory is at the targeted .vcxproj, or you have to specify the target project using `-ProjectName` switch ([reference](https://learn.microsoft.com/en-us/nuget/reference/ps-reference/ps-ref-install-package#examples))..

For now[^1], you can skip unit tests project from the build: Open `OpenGLab.sln` in Visual studio, right click on `UnitTests` project, unload the project, then rebuild the solution.

Best.

[^1]: 3 days before the interviews!

**Reply by MuhammadSulaiman001** [reply]
@Alisl001 I've updated my reply..
